mne/time_frequency/spectrum.py

```diff
@@ -0,0 +1,1186 @@
+        %(cbar_fmt_psd_topo)s
+        %(units_topomap)s
+        %(axes_plot_topomap)s
+        %(show)s
+        %(n_jobs)s
+        %(verbose)s
+        %(method_kw_psd)s
+
+        Returns
+        -------
+        fig : instance of Figure
+            Figure showing one scalp topography per frequency band.
+        """
+        # add after dB
+        # %(sensors_topomap)s
+        # %(show_names_topomap)s
+        # %(mask_evoked_topomap)s
+        # %(mask_params_topomap)s
+        # %(contours_topomap)s
+        # add after sphere
+        # %(image_interp_topomap)s
+        # %(extrapolate_topomap)s
+        # %(border_topomap)s
+        # %(res_topomap)s
+        # %(size_topomap)s
+        # add after vlim
+        # %(colorbar_topomap)s
+        spectrum = self.compute_psd(
+            method=method, tmin=tmin, tmax=tmax, proj=proj,
+            n_jobs=n_jobs, verbose=verbose, **method_kw)
+
+        fig = spectrum.plot_topomap(
+            bands=bands, ch_type=ch_type, normalize=normalize, agg_fun=agg_fun,
+            dB=dB,  # sensors=sensors, show_names=show_names, mask=mask,
+            # mask_params=mask_params, contours=contours,
+            outlines=outlines, sphere=sphere,
+            # image_interp=image_interp, extrapolate=extrapolate,
+            # border=border, res=res, size=size,
+            cmap=cmap, vlim=vlim,  # colorbar=colorbar,
+            cbar_fmt=cbar_fmt, units=units, axes=axes, show=show)
+        return fig
+
+    def _set_legacy_nfft_default(self, tmin, tmax, method, method_kw):
+        """Update method_kw with legacy n_fft default for plot_psd[_topo]().
+
+        This method returns ``None`` and has a side effect of (maybe) updating
+        the ``method_kw`` dict.
+        """
+        if method == 'welch' and method_kw.get('n_fft', None) is None:
+            tm = _time_mask(self.times, tmin, tmax, sfreq=self.info['sfreq'])
+            method_kw['n_fft'] = min(np.sum(tm), 2048)
+
+
+class BaseSpectrum(ContainsMixin, UpdateChannelsMixin):
+    """Base class for Spectrum and EpochsSpectrum."""
+
+    def __init__(self, inst, method, fmin, fmax, tmin, tmax, picks,
+                 proj, reject_by_annotation, *, n_jobs, verbose, **method_kw):
+        # arg checking
+        self._sfreq = inst.info['sfreq']
+        if np.isfinite(fmax) and (fmax > self.sfreq / 2):
+            raise ValueError(
+                f'Requested fmax ({fmax} Hz) must not exceed ½ the sampling '
+                f'frequency of the data ({0.5 * inst.info["sfreq"]} Hz).')
+        _check_option('method', method, ('welch', 'multitaper'))
+
+        # triage method and kwargs. partial() doesn't check validity of kwargs,
+        # so we do it manually to save compute time if any are invalid.
+        psd_funcs = dict(welch=psd_array_welch,
+                         multitaper=psd_array_multitaper)
+        invalid_ix = np.in1d(list(method_kw),
+                             list(signature(psd_funcs[method]).parameters),
```

Comment:
We typically use `_get_args` from `fixes` for this. Can you use it here and/or open an issue that we should investigate whether we can migrate to `inspect.signature` directly nowadays? I can't remember if we just needed to wait until 3.7+ or if `.parameters` isn't enough sometimes, or something else

Reply:
#11089